server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/DeleteTemplateAction.java

```diff
@@ -76,21 +78,50 @@ private void doHandle(DeleteTemplateWsRequest request) {
         request.getTemplateId(), request.getOrganization(), request.getTemplateName()));
       checkGlobalAdmin(userSession, template.getOrganizationUuid());
 
-      checkTemplateUuidIsNotDefault(template.getUuid());
+      DefaultTemplates defaultTemplates = retrieveDefaultTemplates(dbSession, template);
+
+      checkTemplateUuidIsNotDefault(template, defaultTemplates);
       dbClient.permissionTemplateDao().deleteById(dbSession, template.getId());
+      updateViewDefaultTemplateWhenGovernanceIsNotInstalled(dbSession, template, defaultTemplates);
+
       dbSession.commit();
     }
   }
 
+  /**
+   * The default template for view can be removed when Governance is not installed. To avoid keeping a reference
+   * to a non existing template, we update the default templates.
+   */
+  private void updateViewDefaultTemplateWhenGovernanceIsNotInstalled(DbSession dbSession, PermissionTemplateDto template, DefaultTemplates defaultTemplates) {
```

Comment:
what's the relation between fixing organization default templates and deleting a non-default template ?